apps/design/frontend/src/rich_text_editor.tsx

```diff
@@ -0,0 +1,163 @@
+import { useEditor, EditorContent, Editor } from '@tiptap/react';
+import Bold from '@tiptap/extension-bold';
+import Italic from '@tiptap/extension-italic';
+import Underline from '@tiptap/extension-underline';
+import Strike from '@tiptap/extension-strike';
+import Text from '@tiptap/extension-text';
+import Document from '@tiptap/extension-document';
+import Paragraph from '@tiptap/extension-paragraph';
+import HardBreak from '@tiptap/extension-hard-break';
+import BulletList from '@tiptap/extension-bullet-list';
+import OrderedList from '@tiptap/extension-ordered-list';
+import ListItem from '@tiptap/extension-list-item';
+import { Button, ButtonProps } from '@votingworks/ui';
+import styled from 'styled-components';
+
+const StyledEditor = styled.div`
+  cursor: text;
+  border: ${(p) => p.theme.sizes.bordersRem.thin}rem solid
+    ${(p) => p.theme.colors.outline};
+  background: ${(p) => p.theme.colors.containerLow};
+  padding: 0.5rem;
+  line-height: ${(p) => p.theme.sizes.lineHeight};
+  border-radius: ${(p) => p.theme.sizes.borderRadiusRem}rem;
+
+  &:focus-within {
+    background: none;
+    outline: var(--focus-outline);
+  }
+
+  .tiptap {
+    outline: none;
+    padding: 1rem 0 0.5rem;
+
+    > :first-child {
+      margin-top: 0;
+    }
+
+    > :last-child {
+      margin-bottom: 0;
+    }
+
+    li p {
+      margin: 0.25em 0;
+    }
+  }
+`;
+
+const StyledToolbar = styled.div`
+  display: flex;
+  gap: 0.25rem;
+
+  button {
+    padding: 0.25rem 0.5rem;
+  }
+`;
+
+const ControlGroup = styled.div`
+  display: flex;
+  gap: 0.125rem;
+  border: ${(p) => p.theme.sizes.bordersRem.thin}rem solid
+    ${(p) => p.theme.colors.outline};
+  border-radius: ${(p) => p.theme.sizes.borderRadiusRem}rem;
+`;
+
+function ControlButton({
+  isActive,
+  ...props
+}: { isActive: boolean } & ButtonProps) {
+  return (
+    <Button
+      color={isActive ? 'primary' : 'neutral'}
+      fill={isActive ? 'tinted' : 'transparent'}
+      {...props}
+    />
+  );
+}
+
+function Toolbar({ editor }: { editor: Editor | null }) {
+  if (!editor) return null;
```

Comment:
Consider pushing the conditional outside this component:
```suggestion
function Toolbar({ editor }: { editor: Editor }) {
```

Reply:
[b899ddc](https://github.com/votingworks/vxsuite/pull/5359/commits/b899ddc2211ab59ca355a9ba432cb88e236aee76)